### ISSUE TITLE
[reorg] Convert the project to an educational relation (or multi-model) database

### ISSUE THREAD

**Issue by at15** [at15/gegecece#6]
TL;DR 旧坑填不上要开新坑了

The original purpose for gegecece is write implementation based on facebook's gorilla paper. Gorilla is already open sourced and deprecated https://xephonhq.github.io/awesome-time-series-database/database/beringei.html Besides I feel even for tsdb, there is a lot to learn from traditional databases, memory management, query optimization etc. In order to do that I've been spending a lot of time (recently) doing other people's lab, reading blogs and books, however labs are like filling blanks while books sometimes omit details and you can't practice what you learned.

The reasons for choosing relational database (instead of tsdb) are

- libtsdb and xephon are being worked on as well (I didn't abandon my adventure on tsdb), so if there are new tsdb implementation, it will be under those orgs (org that only has me ...)
- There are more resource for relational database, and for OTLP workloads, there are more consistency and optimization solution and problems
- even benchhub need a relational database, it would be good to dog food

The rough roadmap is 

- list basic design requirement and milestones, need to do a survey
- storage engine, prefer LSM over b+, I am not sure if I am able to write a correct (not to mention efficient b+ tree implementation)
- SQL, query optimization
- transaction support, mvcc
- data replication
- distributed transaction

As for language, I might want to follow the multi language model like xephon and reika, go is fast for me to prototype and I am still learning rust & cpp

cc @gaocegege Just FYI, I think you won't mind ... cece: cece is a trade mark of dogfood corp.
